libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -1311,7 +1312,7 @@ void AP_GPS::handle_gps_rtcm_fragment(uint8_t flags, const uint8_t *data, uint8_
     if (len < MAVLINK_MSG_GPS_RTCM_DATA_FIELD_DATA_LEN) {
         rtcm_buffer->fragment_count = fragment+1;
         rtcm_buffer->total_length = (MAVLINK_MSG_GPS_RTCM_DATA_FIELD_DATA_LEN*fragment) + len;
-    } else if (rtcm_buffer->fragments_received == 0x0F) {
```

Comment:
is this right? can't we be missing an earlier one? could be out of order on UDP

Reply:
It's fine either way, we won't flush the buffer until all the fragments have been received, so we must have received the last one before we can test. I did explicitly test this by manually sending packets up out of order from a GCS and I had instrumented the state to see where we were and it behaved correctly.